app/packages/core/src/components/Actions/similar/Similar.tsx

```diff
@@ -82,8 +84,11 @@ const SortBySimilarity = ({
   const isLoading = useRecoilValue(fos.similaritySorting);
 
   useLayoutEffect(() => {
-    choices.choices.length === 1 &&
-      updateState({ brainKey: choices.choices[0] });
+    if (!choices.choices.includes(state.brainKey)) {
+      const newKey =
+        choices.choices.length > 0 ? choices.choices[0] : undefined;
+      updateState({ brainKey: newKey });
+    }
   }, [choices]);
```

Comment:
Should `state.brainKey` not be included in the dependency array, or is it intentional? If it's intentional (IDK why), perhaps we should make it clear in a comment. I'd also add `updateState` in the dep list to be exhaustive, although that's referentially stable.

Reply:
Thanks for catching this. It should be included.